scripts/ocamldeps.sh

```diff
@@ -0,0 +1,243 @@
+      } else {
+        implementation_provides[unit]=$4
+      }
+    } else {
+      printf "WARN: state %s, expected cmx, got %s\n", state, $0 > "/dev/stderr"
+    }
+    state="crc"
+    next
+  }
+
+  /^Interfaces imported:/ {
+    state="interface"
+    next
+  }
+
+  /^Implementations imported:/ {
+    state="implementation"
+    next
+  }
+
+  /^\t/ {
+    if (state == "interface" && NF > 1 && match($1, "^-") == 0) {
+      if (unit == $2) {
+        if (ignore_interface[unit] != "") {
+          if (ignore_interface[unit] != "seen") {
+            printf "INFO: ignoring Provides %s(%s)=%s from %s\n", rpm_prefix_interface, unit, $1, file  > "/dev/stderr"
+            ignore_interface[unit]="seen"
+          }
+        } else {
+          interface_provides[unit]=$1
+        }
+      } else {
+        if (ignore_interface[$2] != "") {
+          if (ignore_interface[$2] != "seen") {
+            printf "INFO: ignoring Requires %s(%s)=%s from %s\n", rpm_prefix_interface, $2, $1, file  > "/dev/stderr"
+            ignore_interface[$2]="seen"
+          }
+        } else {
+          interface_requires[$2]=$1
+        }
+      }
+      next
+    } else if (state == "implementation" && NF > 1 && match($1, "^-") == 0) {
+      if (unit == $2) {
+        if (ignore_implementation[unit] != "") {
+          if (ignore_implementation[unit] != "seen") {
+            printf "INFO: ignoring Provides %s(%s)=%s from %s\n", rpm_prefix_implementation, unit, $1, file  > "/dev/stderr"
+            ignore_implementation[unit]="seen"
+          }
+        } else {
+          implementation_provides[unit]=$1
+        }
+      } else {
+        if (ignore_implementation[$2] != "") {
+          if (ignore_implementation[$2] != "seen") {
+            printf "INFO: ignoring Requires %s(%s)=%s from %s\n", rpm_prefix_implementation, $2, $1, file  > "/dev/stderr"
+            ignore_implementation[$2]="seen"
+          }
+        } else {
+          implementation_requires[$2]=$1
+        }
+      }
+      next
+    } else  {
+      next
+    }
+  }
+  /^.*/ {
+    state="find"
+  }
+
+  END {
+    if (mode == "prov") {
+      for (i in interface_provides) {
+        printf "%s(%s) = %s\n", rpm_prefix_interface, i, interface_provides[i]
+      }
+      for (i in implementation_provides) {
+        printf "%s(%s) = %s\n", rpm_prefix_implementation, i, implementation_provides[i]
+      }
+    }
+    if (mode == "req") {
+      for (i in interface_requires) {
+        printf "%s(%s) = %s\n", rpm_prefix_interface, i, interface_requires[i]
+      }
+      for (i in implementation_requires) {
+        printf "%s(%s) = %s\n", rpm_prefix_implementation, i, implementation_requires[i]
+      }
+    }
+  }
+  '
+}
+#
+#
+usage() {
+    echo >&2 "Usage: ${0##*/} -prov|-req [-f 'ocamlobjinfo cmd']"
+}
+#
+mode=
+ignore_implementation_a=()
+ignore_interface_a=()
+while test "$#" -gt 0
+do
+  : "${1}" "${2}"
+  case "${1}" in
+    -prov) mode='prov' ;;
+    -req) mode='req' ;;
```

Comment:
Shouldn't this be something like the following?

```bash
    -P|--provides) mode='prov' ;;
    -R|--requires) mode='req' ;;
```

Reply:
Why? It is an internal helper, not a public interface.

Reply:
Because people do directly run it when doing things like debugging or testing them. And it's just _weird_ to ignore conventions for the parameterization of a dep generator.